FCP-INDI/C-PAC
# Merge feature/applywarp_parallel into develop branch

## Fixes


Related to #1341 by @sgiavasis 

## Description
Parallelizes warp applications for time series data.

## Technical details
Triggers the creation of a Map Node for the apply warp nodes - usually these are only set as Map Nodes when applying warps to 3D functional outputs (i.e. derivatives) that have many versions due to multiple ROIs- so basically, dual regression and the SCA outputs.

But this time, the Map Node is created for the multiple slices of the time series:
https://github.com/FCP-INDI/C-PAC/blob/feature/applywarp_parallel/CPAC/registration/output_func_to_standard.py#L485-L487

## Tests
- Run preprocessing with multiple CPUs per participant and ensure the time series (`functional_to-standard`) is broken up, registered, and concatenated back together properly.
- Run preprocessing with only one CPU per participant for two reasons:
    - To make sure it still works serially!
    - To compare the `functional_to_standard` output with the output of the multiple CPU run.

## Checklist

  index.md`).
  visible errors.

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
